### PR TITLE
CBG-464: Extend go-fleecedelta to take interface{} for patching

### PR DESCRIPTION
**Pull request by JRascagneres**
- [x] Merge https://github.com/couchbaselabs/go-fleecedelta/pull/24

### CODE REVIEW

File: base/deltas_ee.go [hunk 1]
```diff
@@ -26,3 +26,8 @@ func Diff(old, new map[string]interface{}) (delta []byte, err error) {
 func Patch(old *map[string]interface{}, delta []byte) (err error) {
 	return fleecedelta.PatchJSON(old, delta)
 }
+
+// PatchJSONWithMap attempts to path old with the given delta passed as a map[string]interface{}
+func PatchMap(old *map[string]interface{}, delta map[string]interface{}) (err error) {
```
Comment (bbrks) [inline base/deltas_ee.go]: My vote is to replace the existing Patch function the one that takes `interface{}`. There's only one other instance of `Patch()` being used (in the BLIP test client), and that's doing the same marshal -> unmarshal work that we're trying to remove.

Let's try and keep the SG side of this minimal, as we're not concerned about other types of Patching here.

**Review by bbrks — CHANGES_REQUESTED** [review]

File: base/deltas_ee.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ func Diff(old, new map[string]interface{}) (delta []byte, err error) {
 	return fleecedelta.DiffJSON(old, new)
 }
 
-// Patch will patch old with the given delta.
-func Patch(old *map[string]interface{}, delta []byte) (err error) {
-	return fleecedelta.PatchJSON(old, delta)
+// Patch attempts to patch old with the given delta passed as a map[string]interface{}
+func Patch(old *map[string]interface{}, delta map[string]interface{}) (err error) {
```
Comment (bbrks) [inline base/deltas_ee.go:26]: ~What about the `interface{}` part of the previous comment?~

File: base/deltas_ce.go [hunk 3]
```diff
@@ -14,7 +14,7 @@ func Diff(old, new map[string]interface{}) (delta []byte, err error) {
 	return nil, ErrDeltasNotSupported
 }
 
-// Patch is only implemented in EE, the CE stub always returns an error.
-func Patch(old *map[string]interface{}, delta []byte) (err error) {
+// PatchJSONWithMap attempts to patch old with the given delta passed as a map[string]interface{}
```
Comment (bbrks) [inline base/deltas_ce.go]: Comment doesn't match function.

**Review by bbrks — CHANGES_REQUESTED** [review]

**Review by adamcfraser — APPROVED** [review]

File: base/deltas_ce.go [hunk 4]
```diff
@@ -14,7 +14,7 @@ func Diff(old, new map[string]interface{}) (delta []byte, err error) {
 	return nil, ErrDeltasNotSupported
 }
 
-// Patch is only implemented in EE, the CE stub always returns an error.
-func Patch(old *map[string]interface{}, delta []byte) (err error) {
+// Patch attempts to patch old with the given delta passed as a map[string]interface{}
```
Comment (bbrks) [inline base/deltas_ce.go]:
```suggestion
// Patch is only implemented in EE, the CE stub always returns an error.
```

**Review by bbrks — CHANGES_REQUESTED** [review]
@JRascagneres sorry, one last comment - just commit it through GitHub's web UI and I'll merge.

**Review by bbrks — APPROVED** [review]

**Pull request merged**